### DISCUSSION TITLE
How can I reduce four convert commands to one?

### DISCUSSION THREAD

**Discussion by J1STAR** [ImageMagick/ImageMagick#5691] [Help] [answered]
command 1: convert profile.png -resize 300x300 -size 300x300 xc:white +swap -gravity center -composite profile_resize.png
command 2: convert source1.png -resize 150x150 -size 150x150 xc:white +swap -gravity center -composite source1_resize.png
command 3: convert -stroke "#EEEEEE" -strokewidth 5 -size 150x150 xc:none -fill source1_resize.png -draw "circle 75,75 75,15" source1_resize_with_border.png
command 4: convert profile_resize.png source1_resize_with_border.png -geometry 150x150+75+75 -composite thumbnail.png

profile.png
![01_NAVER_Logo_Green](https://user-images.githubusercontent.com/8870540/197392102-b3d51d3d-b47a-48cf-bb80-50e4c1a5c6bc.png)

source1.png
![ic_channel_chat](https://user-images.githubusercontent.com/8870540/197392006-31350dae-85df-4723-b30f-ab4f96924238.png)

what i want
![01_NAVER_Logo_Green](https://user-images.githubusercontent.com/8870540/197392000-24e7afe8-f5a5-45d7-8f8d-21471ba9f1b4.png)

i tried this command 
convert \\( profile.png -resize 300x300 -size 300x300 xc:white +swap -gravity center -composite \\) \\
             \\( \\( source1.png -resize 120x120 -size 150x150 xc:white +swap -gravity center -composite \\) \\
                \\( -stroke "#EEEEEE" -strokewidth 5 -size 150x150 xc:black -fill white -draw "circle 75,75 75,15" -alpha copy \\) \\
            -compose copy_opacity -composite \\) \\
            -geometry 150x150+75+75 -compose copy_opacity -composite thumbnail.png

result
![thumbnail](https://user-images.githubusercontent.com/8870540/197392740-8e3037e6-1404-46bd-85dc-0d2ed3b3d2d1.png)

this confuses me
I still don't understand😢

**Comment by snibgo** [comment]
Why have you added "-compose copy_opacity "?

EDIT to add: Your four commands don't give you what you say you want.

**Reply by J1STAR** [reply]
I used it because the background of the converted source1 was white.
thx

**Comment by J1STAR** [comment, marked as answer]
answering own question
convert \\( profile.png -resize 300x300 -size 300x300 xc:white +swap -gravity center -composite \\) \\
\\( \\( source1.png -resize 120x120 -size 150x150 xc:white +swap -gravity center -composite -write mpr:resize_source \\) \\
    \\( -stroke "#EEEEEE" -strokewidth 5 -size 150x150 xc:transparent -fill mpr:resize_source -draw "circle 75,75 75,2.5" \\) \\
-delete 0,2 -write mpr:sub_thumb \\) \\
-geometry 150x150+75+75 -composite thumbnail.png

thx
